### DISCUSSION TITLE
record username(or email) in the admin UI request log

### DISCUSSION THREAD

**Discussion by xu4wang** [pocketbase/pocketbase#4746] [Q&A] [answered]
For monitoring purpose, I'm thinking to record the people who is initiating the request.

Is there a way to record `pb.authStore.baseModel.username` into the log database?

**Comment by ganigeorgiev** [comment, marked as answer]
No, it is not possible at the moment (_I've added it in my todo to consider eventually allowing access to the request context to the logs with the planned refactoring_).

**Reply by xu4wang** [reply]
Thanks for the answer.

Is there a easy way to record username into log database? can I make a hook or middleware to do that?

**Reply by xu4wang** [reply]
I guess I can modify the ActivityLogger https://github.com/pocketbase/pocketbase/blob/34a25f640b8d2d041bf628e1e11ee2d11eb29083/apis/middlewares.go#L286 for that.  But I'm using pocketbase official release, only extending it via JS hooks for now.

**Reply by ganigeorgiev** [reply]
Depending on your use case a workaround could be to send the username/email as `Referer` header either with the [JS SDK `beforeSend` hook](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#send-hooks) before sending the requests or via a global middleware that will change the request header based on the loaded auth context. Here is one hacky example using the JS `pb_hooks` (I haven't tested it):

```js
// pb_hooks/main.pb.js

routerUse((next) => {
    return (c) => {
        // https://pocketbase.io/docs/js-routing/#retrieving-the-current-auth-state
        const authRecord = c.get("authRecord")
        if (authRecord) {
            c.request().header.set("Referer", authRecord.get("username"))
        }

        return next(c)
    }
})
```

**Reply by xu4wang** [reply]
I'm using below beforeSend hook.  it's tested working. thanks. @ganigeorgiev 


```javascript
    let userEmail = 'GUEST';

    try {
        userEmail = locals.pb.authStore.baseModel.email;
    } catch (_) {
    }

    locals.pb.beforeSend = function (url, options) {
        options.headers.Referer = userEmail;
        return { url, options };
    };

```
